static/app/components/core/layout/stack.mdx

```diff
@@ -0,0 +1,243 @@
+---
+title: Stack
+description: A simplified layout component built on Flex that provides easy vertical stacking with responsive props and spacing controls.
+source: 'sentry/components/core/layout/stack'
+resources:
+  js: https://github.com/getsentry/sentry/blob/master/static/app/components/core/layout/stack.tsx
+---
+
+import {Container, Stack} from 'sentry/components/core/layout';
+import * as Storybook from 'sentry/stories';
+
+import APIReference from '!!type-loader!sentry/components/core/layout/stack';
+
+export const types = {Stack: APIReference.Stack};
+
+The `Stack` component is a simplified layout component built on top of the `Flex` component. It provides a focused API for common stacking layouts with only the essential props: `direction`, `align`, `justify`, and `gap`. By default, Stack uses column direction making it perfect for vertical layouts.
+
+## Basic Usage
+
+To create a basic vertical stack, wrap elements in `<Stack>` and they will be laid out vertically using flexbox.
+
+```jsx
+<Stack>
+  <div>Item 1</div>
+  <div>Item 2</div>
+  <div>Item 3</div>
+</Stack>
+```
+
+### When to Use Stack vs Flex
+
+Use `Stack` when you need a simple, focused layout component for common stacking patterns. It's perfect for:
+
+- Vertical lists or forms
+- Simple horizontal arrangements
+- Basic responsive layouts
+
+Use `Flex` when you need the full power of flexbox with properties like `wrap`, `flex`, `inline`, or complex alignment scenarios.
+
+### Composition
+
+The `Stack` implements composition via <a href="/stories/layout/composition">render prop</a> pattern.
+
+<Storybook.Demo>
+  <Stack
+    border="primary"
+    radius="md"
+    padding="md"
+    justify="between"
+    background="primary"
+    width="80%"
+    gap="md"
+  >
+    {props => (
+      <div {...props}>
```

Comment:
Imo this example makes little sense because `Stack` is just a `div`. This is exactly what we have in the composition docs about [when not to use render props](https://sentry.io/stories/layout/composition#when-not-to-use-render-props) 😅

Reply:
Changed it to use a CustomComponent :)! Good eye!